### PR TITLE
Shadowkin Rework

### PR DESCRIPTION
**Pull request by FoxxoTrystan**
# Description

This PR compleatly rework shadowkins, slightly been forced this makes shadowkins more "psionic creature" then anything else.
This is a WIP, but in general this PR make Shadowkin "Psionic" and fall under the same rule.

RIP Shadowkin for me but ye.

CHANGES TO SHADOWKINS:
Shadowkins now follow the laws of psionic, they will always spawn with DarkSwapPower but can generate others power, they do not spawn with Shadeskip by default anymore, Mana has been lower from 250 to 200; they can now pick normal psionic jobs.

Changes to DarkSwap:
DarkSwap cost 30 Mana and will raise glimmer on use, its will drain 1 Mana/Sec, will eject the user when mana is fully drained.
DarkSwap is now a random roll.

Changes to Psionics:
Added the Mana Alert for all psionics

---

# TODO

- [x] Psionic Mana Alert
- [x] DarkSwap Rework
- [x] Remove Unique Shadowkin Psionic Changes
- [x] Import bugs fixes/QOL changes.

---

<details><summary><h1>Media</h1></summary>
<p>

N/A

</p>
</details>

---

# Changelog

:cl:
- tweak: Shadowkins has returned and has been reworked.
- add: DarkSwap is now in the Psionic Power Pool.
- tweak: Psionics can now see their mana pools.

### CODE REVIEW

**Comment by FoxxoTrystan** [comment]
PR Ready for review.

File: Content.Server/Shadowkin/ShadowkinSystem.cs [hunk 1]
```diff
@@ -166,16 +118,13 @@ private void OnRejuvenate(EntityUid uid, ShadowkinComponent component, Rejuvenat
...
         EnsureComp<PsionicComponent>(uid, out var magic);
-        magic.Mana = 250;
-        magic.MaxMana = 250;
+        magic.Mana = 200;
+        magic.MaxMana = 200;
         magic.ManaGain = 0.25f;
-        magic.BypassManaCheck = true;
-        magic.Removable = false;
         magic.MindbreakingFeedback = "shadowkin-blackeye";
+        magic.NoMana = "shadowkin-tired";
```
Comment (DEATHB4DEFEAT) [inline Content.Server/Shadowkin/ShadowkinSystem.cs:125]: I think you can do this with something like this
```cs
if (!HasComp<PsionicComponent>(uid))
	AddComp(uid, new PsionicComponent
		{
			Mana = 200;
			MaxMana = 200;
			// etc...
		});
```

File: Content.Shared/Shadowkin/EtherealComponent.cs [hunk 2]
```diff
@@ -23,8 +23,22 @@ public sealed partial class EtherealComponent : Component
     [DataField]
     public float DarkenRate = 0.084f;
 
+    /// <summary>
+    ///     Can be stunned by ethereal stun objects?
+    /// </summary>
```
Comment (DEATHB4DEFEAT) [inline Content.Shared/Shadowkin/EtherealComponent.cs:28]:
```suggestion
    /// Can this be stunned by ethereal stun objects?
```

File: Content.Shared/Shadowkin/EtherealComponent.cs [hunk 3]
```diff
@@ -23,8 +23,22 @@ public sealed partial class EtherealComponent : Component
...
+    [DataField]
+    public bool CanBeStunned = true;
+
+    /// <summary>
+    ///     Drain Mana is ent is psionic?
+    /// </summary>
```
Comment (DEATHB4DEFEAT) [inline Content.Shared/Shadowkin/EtherealComponent.cs:34]:
```suggestion
    /// Drain Mana if this entity is psionic?
```

File: Resources/Locale/en-US/psionics/psionic-powers.ftl [hunk 4]
```diff
@@ -167,13 +167,11 @@ psionic-roll-failed = For a moment, my consciousness expands, yet I feel that it
...
 
-# Shadowkin ShadeSkip
-action-description-shadowkin-shadeskip = Aaramrra!
-
 # DarkSwap
 action-name-darkswap = DarkSwap
 action-description-darkswap = Mmra Mamm!
-
+darkswap-power-initialization-feedback =
+    For a short moment, i find myself able to break the realms... phasing in the shadows and the dark, like shadowkins... ready to traval to the darknest of places...
```
Comment (DEATHB4DEFEAT) [inline Resources/Locale/en-US/psionics/psionic-powers.ftl:174]:
```suggestion
    For a short moment, I find myself able to break the realms. Phasing in the shadows and the dark, ready to travel to the darkest of places...
```

**Review by DEATHB4DEFEAT — APPROVED** [review]

File: Content.Server/Shadowkin/ShadowkinSystem.cs (hunk 1 above)
Comment (FoxxoTrystan) [reply Content.Server/Shadowkin/ShadowkinSystem.cs:125]: Oh Neat

**Review by sleepyyapril — APPROVED** [review]

**Pull request merged**